airflow/contrib/hooks/gcp_cloudml_hook.py

```diff
@@ -0,0 +1,139 @@
+from airflow.contrib.hooks.gcp_api_base_hook import GoogleCloudBaseHook
+from airflow import settings
+
+logging.getLogger('GoogleCloudML').setLevel(settings.LOGGING_LEVEL)
+
+
+class _CloudMLJob(object):
+    """CloudML job operations helper class."""
+
+    def __init__(self, cloudml, project_name, job_id, job_spec=None):
+        assert project_name
+        self._cloudml = cloudml
+        self._project_name = 'projects/{}'.format(project_name)
+        self._job_id = job_id
+
+        assert self._job_id
+        self._job_spec = job_spec
+        if self._job_spec:
+            assert self._job_id == self._job_spec['jobId']
+
+    def get_job(self):
+        """Gets a CloudML job based on the job name.
+
+        :return: CloudML job object if succeed.
+        :rtype: dict
+        """
+        name = '{}/jobs/{}'.format(self._project_name, self._job_id)
+        request = self._cloudml.projects().jobs().get(name=name)
+        while True:
+            try:
+                return request.execute()
+            except errors.HttpError as e:
+                if e.resp.status == 429:
+                    time.sleep(10)  # polling after 10 seconds
+                else:
+                    logging.error('Failed to get CloudML job: {}'.format(e))
+                    raise
+
+    def create_job(self):
+        """Creates a Job on Cloud ML.
+
+        :return: CloudML job creation request response.
+        :rtype: dict
+        """
+        request = self._cloudml.projects().jobs().create(
+            parent=self._project_name, body=self._job_spec)
+        try:
+            return request.execute()
+        except errors.HttpError as e:
+            logging.error('Failed to create CloudML job: {}'.format(e))
+            raise
+
+    def wait_for_done(self, interval):
+        """Waits for the Job to reach a terminal state.
+
+        This method will periodically check the job state until the job reach
+        a terminal state.
+
+        :param interval: Polling interval in seconds.
+        :return: CloudML job object if succeed.
+        :rtype: dict
+        """
+        assert interval > 0
+        while True:
+            job = self.get_job()
+            state = job['state']
+            if state in ['FAILED', 'SUCCEEDED', 'CANCELLED']:
+                return job
+            time.sleep(interval)
+
+
+class CloudMLHook(GoogleCloudBaseHook):
+
+    def __init__(self, gcp_conn_id='google_cloud_default', delegate_to=None):
+        super(CloudMLHook, self).__init__(gcp_conn_id, delegate_to)
+        self._cloudml = self.get_conn()
+
+    def get_conn(self):
+        """
+        Returns a Google CloudML service object.
+        """
+        credentials = GoogleCredentials.get_application_default()
+        return build('ml', 'v1', credentials=credentials)
+
+    def create_job(self, project_name, job):
+        """
+        Creates a CloudML Job, and returns the Job object, which can be waited
```

Comment:
Thanks for the catch! We wrote a different docstring for that. Hopefully it reveals clearly what it does.